### PR TITLE
feat: Add automatic standard CI variable resolution

### PR DESCRIPTION
**Pull request by aaronsteers**
# feat: Add automatic standard CI variable resolution

## Summary

This PR implements automatic resolution of standard CI variables from GitHub context, eliminating the need for complex expressions like `${{ github.event.pull_request.head.repo.full_name || github.repository }}` throughout workflows.

**Key Changes:**
- Added `standard_ci_vars` input parameter (default: false) to enable the feature
- Implements resolution for 20+ standard variables: `pr-number`, `head-ref`, `base-ref`, `repo`, `head-repo`, `base-repo`, etc.
- Smart `gitref` resolution handles "run from main, operate on PR branch" pattern for slash commands
- Fork-aware: `repo` variables point to base repository, `gitref` points to fork's branch
- Issue-aware: Git refs are null for issue events since issues don't have branches
- Maintains full backward compatibility with existing functionality

## Review & Testing Checklist for Human

- [ ] **Test with actual fork PRs** - Verify `repo` resolves to base repo while `gitref` resolves to fork's branch name
- [ ] **Test slash commands from different contexts** - Verify workflow_dispatch with/without `pr` input resolves variables correctly  
- [ ] **Test issue vs PR trigger scenarios** - Confirm git refs are null for issue events but populated for PR events
- [ ] **Verify backward compatibility** - Run existing workflows with `standard_ci_vars: false` to ensure no regressions
- [ ] **Test edge cases** - Missing PR data, gh CLI unavailable, workflow_dispatch without clear context

**Recommended Test Plan:**
1. Create a fork PR and trigger workflows to verify repo vs gitref resolution
2. Test slash commands (`/test`, etc.) to ensure smart gitref fetches correct branch
3. Run action on issue comments to verify git refs are properly nulled
4. Test with existing workflows to confirm no breaking changes

---

### Diagram

```mermaid
%%{ init : { "theme" : "default" }}%%
graph TD
    A["action.yml"]:::major-edit --> B["resolve-standard-ci-vars step"]:::major-edit
    A --> C["merge-results step"]:::minor-edit
    D["README.md"]:::major-edit --> E["Standard CI Variables table"]:::major-edit
    F[".github/workflows/test-action.yml"]:::major-edit --> G["test-standard-ci-vars job"]:::major-edit
    
    B --> H["GitHub Context<br/>(github.event.*)"]:::context
    B --> I["Smart gitref resolution<br/>(gh pr view)"]:::context
    C --> J["Final JSON output<br/>(all variables)"]:::context
    
    subgraph Legend
        L1[Major Edit]:::major-edit
        L2[Minor Edit]:::minor-edit  
        L3[Context/No Edit]:::context
    end

    classDef major-edit fill:#90EE90
    classDef minor-edit fill:#87CEEB
    classDef context fill:#FFFFFF
```

### Notes

- The smart gitref resolution using `gh pr view` has fallback logic but may need testing in constrained CI environments
- Fork handling logic is critical - ensure `repo` (base) vs `gitref` (head) distinction works correctly
- Requested by AJ Steers (@aaronsteers)  
- Session: https://app.devin.ai/sessions/2ea6e9db29ce437d8ca81c377c8660b8
- Link to Devin run: https://app.devin.ai/sessions/2ea6e9db29ce437d8ca81c377c8660b8

### CODE REVIEW

File: .github/workflows/test-action.yml [hunk 1]
```diff
@@ -94,3 +94,136 @@ jobs:
...
+            exit 1
+          fi
+
+          # Ensure custom variables still work
+          if [ "${{ fromJSON(steps.test-no-ci-vars.outputs.all).test_var }}" != "backward_compat" ]; then
+            echo "ERROR: custom variables should still work when standard_ci_vars is false"
+            exit 1
+          fi
+
+          echo "✅ Backward compatibility test passed!"
```
Comment (aaronsteers) [inline .github/workflows/test-action.yml:229]: This step can then be renamed to just 'Test with standard CI vars disabled'. (I don't like "backwards compatibility" as an evergreen name.)

File: action.yml [hunk 2]
```diff
@@ -23,10 +23,106 @@ inputs:
       Alias for log_outputs (true = show evaluated output)
     default: 'false'
     required: false
+  standard_ci_vars:
+    description: >
+      Whether to automatically resolve standard CI variables from GitHub context.
+      When enabled, variables like pr-number, source-branch, target-branch, etc. are automatically
+      populated based on the trigger type.
+    default: 'true'
+    required: false
```
Comment (aaronsteers) [inline action.yml:32]: Let's actually just delete this input option. These vars will always be set.

File: action.yml [hunk 3]
```diff
@@ -122,6 +234,186 @@ runs:
...
+          CI_VARS["pr-target-git-tag"]=""
+        fi
+        
+        # Smart resolution for workflow_dispatch with PR input
+        if [[ "${{ github.event_name }}" == "workflow_dispatch" && -n "${{ inputs.pr }}" ]]; then
+          # Validate PR input to prevent command injection
+          pr_input="${{ inputs.pr }}"
+          if [[ "$pr_input" =~ ^[0-9]+$ ]] || [[ "$pr_input" =~ ^https://github\.com/[^/]+/[^/]+/pull/[0-9]+$ ]]; then
+            echo "Resolving PR data for PR $pr_input..."
+            if command -v gh >/dev/null 2>&1; then
```
Comment (aaronsteers) [inline action.yml:333]: Move the check to confirm gh is available into its own step.
